examples/cookbook/app/network-requests/PhoneBook.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import { Text } from 'react-native';
+import { User } from './types';
+import ContactsList from './components/ContactsList';
+import FavoritesList from './components/FavoritesList';
+import getAllContacts from './api/getAllContacts';
+import getAllFavorites from './api/getAllFavorites';
+
+export default () => {
+  const [usersData, setUsersData] = useState<User[]>([]);
+  const [favoritesData, setFavoritesData] = useState<User[]>([]);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
```

Comment:
thinking out loud, perhaps we should showcase usage with TanStack Query instead of manual promise fetching. Implementing data fetching using useEffect to avoid race conditions is [verbose](https://react.dev/reference/react/useEffect#fetching-data-with-effects). Wdyt @vanGalilea ?

Reply:
Showcase is a very good idea, as it is become and industry standard the last years.
With my approach I intended to keep things more straight-forward.

Reply:
Let me add it up, this week 👍🏻 In the meanwhile, you can review the rest ;)